src/user.cpp

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2024 <mbernard>************************************************* */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   user.cpp                                           :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: mbernard <mbernard@student.42lyon.fr>      +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/10/24 09:46:04 by mbernard          #+#    #+#             */
+/*   Updated: 2024/10/29 15:13:42 by mbernard         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "../includes/Parser.hpp"
+
+static std::vector<std::string> fillUserVectorString(const std::string str) {
+  std::vector<std::string> result;
+  std::istringstream iss(str);
+  std::string token;
+  int8_t counter = 0;
+
+  while (iss >> token && counter < 4) {
+    result.push_back(token);
+    ++counter;
+  }
+  if (counter < 4 || result.at(3)[0] != ':') {
+    result.clear();
+    return (result);
+  }
+  std::string lastToken = str.substr(str.find_first_of(":"));
+  if (lastToken.empty() || lastToken.size() > 100)
+    result.clear();
+  else
+    result.at(3) = lastToken;
+  return (result);
+}
+
+static bool usernameContainsForbiddenCaracters(const std::string username) {
+  size_t size = username.size();
+
+  for (size_t i = 0; i < size; ++i) {
+    if (std::isspace(username.at(i)) || std::iscntrl(username.at(i))) {
+      std::cout << "space or control char" << std::endl;
+    // ERR_ERRONEUSNICKNAME (432)
+      return (true);
+    }
+  }
+  return (false);
+}
+
+static bool realnameContainsForbiddenCaracters(const std::string realname) {
+  size_t size = realname.size();
+
+  for (size_t i = 0; i < size; ++i) {
+    if (std::iscntrl(realname.at(i))) {
+      std::cout << "control char" << std::endl;
+    // ERR_ERRONEUSNICKNAME (432)
+      return (true);
+    }
+  }
+  return (false);
+}
+
+bool Parser::verifyUser(std::string user, Client& client, clientsMap cltMap) {
```

Comment:
Need to check if the client is already registered. If yes, the server should send 462 numeric reply.